### PR TITLE
ROSA: Large/Medium Scale Testing

### PR DESCRIPTION
**Pull request by krishvoor**
This PR introduces large and medium testing for node-density-cni and node-density

### CODE REVIEW

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-249nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-249nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-24nodes

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-249nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-heavy-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-heavy-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-cni-120nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-heavy-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-heavy-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-cni-120nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 1]
```diff
@@ -113,26 +113,64 @@ tests:
...
     workflow: rosa-aws-sts
   timeout: 5h0m0s
+- as: node-density-cni-249nodes
+  cron: 0 12 4 * *
+  steps:
+    allow_skip_on_success: true
+    cluster_profile: aws-perfscale
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.2xlarge
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:131]: m5.xl pls

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 2]
```diff
@@ -113,26 +113,64 @@ tests:
...
+    workflow: rosa-aws-sts
+  timeout: 7h0m0s
+- as: node-density-cni-120nodes
+  cron: 0 18 4 * *
+  steps:
+    allow_skip_on_success: true
+    cluster_profile: aws-perfscale
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.2xlarge
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:150]: m5.xl pls

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 3]
```diff
@@ -113,26 +113,64 @@ tests:
...
+  timeout: 7h0m0s
 - as: node-density-cni-24nodes
   cron: 0 15 * * 4
   steps:
     allow_skip_on_success: true
     cluster_profile: aws-perfscale-qe
     env:
       CHANNEL_GROUP: nightly
-      COMPUTE_MACHINE_TYPE: m5.xlarge
+      COMPUTE_MACHINE_TYPE: m5.2xlarge
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:169]: m5.xl

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml [hunk 4]
```diff
@@ -111,24 +111,60 @@ tests:
...
+    workflow: rosa-aws-sts
+  timeout: 8h0m0s
+- as: node-density-cni-120nodes
+  cron: 0 18 5 * *
+  steps:
+    allow_skip_on_success: true
+    cluster_profile: aws-perfscale
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.xlarge
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:145]: m5.xl pls

**Review by afcollins — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml]: Ack, updated

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml (hunk 4 above)
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:145]: Ack, updated

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-cni-120nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-249nodes

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-cni-120nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-249nodes

File: ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml [hunk 5]
```diff
@@ -11,7 +11,7 @@ ref:
     mount_path: /secret
   env:
   - name: E2E_VERSION
-    default: "v2.1.4"
+    default: "kube-burner-ocp"
```
Comment (venkataanil) [inline ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:14]: New E2E is released and this branch got deleted. We need to use v2.1.5 E2E_VERSION

Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:14]: Ack, updating this.

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-node-density-cni-120nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-cni-120nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 6]
```diff
@@ -46,7 +46,7 @@ tests:
       OPENSHIFT_VERSION: "4.14"
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:46]: PROFILE_TYPE: "reporting" needed here.

File: ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml [hunk 7]
```diff
@@ -11,7 +11,7 @@ ref:
     mount_path: /secret
   env:
   - name: E2E_VERSION
-    default: "v2.1.4"
+    default: "v2.1.5"
```
Comment (afcollins) [inline ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:14]: Do we want to be incrementing e2e version inconsistently across the workloads? cdv2, node-density, etc. will all be on v2.1.4 with this change.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml [hunk 8]
```diff
@@ -46,7 +46,7 @@ tests:
       OPENSHIFT_VERSION: "4.15"
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:46]: Again, do we want report profile for the 249node test?

File: ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml (hunk 7 above)
Comment (afcollins) [reply ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:14]: Oh. I see 2.1.5 updated consistently in the other PR.

**Comment by krishvoor** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-control-plane-249nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-control-plane-249nodes

**Comment by krishvoor** [comment]
The last [4.15](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/47827/rehearse-47827-periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-control-plane-249nodes/1750575169518178304) failure was not related to PR, instead was a installation-failure

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml (hunk 6 above)
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:46]: Added - ptal

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-node-density-cni-24nodes

**Comment by venkataanil** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-control-plane-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-control-plane-24nodes

File: ci-operator/step-registry/openshift-qe/node-density-heavy/openshift-qe-node-density-heavy-ref.yaml
Comment (venkataanil) [inline ci-operator/step-registry/openshift-qe/node-density-heavy/openshift-qe-node-density-heavy-ref.yaml:1]: change default PODS_PER_NODE value to 200

File: ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml
Comment (venkataanil) [inline ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:1]: change default PODS_PER_NODE value to 200.
Can you please also change default PODS_PER_NODE value to 245 in openshift-qe-node-density-ref.yaml file?
These are the recommended values in our ADR.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml (hunk 8 above)
Comment (venkataanil) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:46]: metrics-aggregated.yml or metrics.yml may timeout (and index large number of docs to ES) in 249 node environment, so better we can use report mode.

Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:46]: Ack, updated - ptal

File: ci-operator/step-registry/openshift-qe/node-density-heavy/openshift-qe-node-density-heavy-ref.yaml
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/node-density-heavy/openshift-qe-node-density-heavy-ref.yaml:1]: ptal here: https://github.com/openshift/release/pull/48158

File: ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:1]: Done, ptal https://github.com/openshift/release/pull/48158

Comment (afcollins) [reply ci-operator/step-registry/openshift-qe/node-density-cni/openshift-qe-node-density-cni-ref.yaml:1]: We run the tests at 200 when we want to compare, but I don't want to set the defaults to 200. Default of 100 is a more comfortable value for adding new tests so the test authors can focus on making sure the test plumbing. Then, once the job is good, they set that specific job to 200ppn.

That was my idea around it at least.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml (hunk 8 above)
Comment (venkataanil) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:46]: I agree that we don't want report profile for 4.15. However metrics-aggregated.yml and metrics.yml getting timeoutd on large scale. So this is the need of the hour.
We will remove report profile on 4.15 once https://github.com/kube-burner/kube-burner-ocp/pull/3 merged and similar change available in metrics.yml

**Review by venkataanil — COMMENTED** [review]
I agree that we don't want report profile for 4.15. However metrics-aggregated.yml and metrics.yml getting timeoutd on large scale. So this is the need of the hour.
We will remove report profile on 4.15 once https://github.com/kube-burner/kube-burner-ocp/pull/3 merged and similar change available in metrics.yml

**Review by venkataanil — APPROVED** [review]
LGTM

**Comment by krishvoor** [comment]
@afcollins can you ptal?

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 9]
```diff
@@ -31,8 +31,8 @@ resources:
       cpu: 100m
       memory: 200Mi
 tests:
-- as: cluster-density-v2-249nodes
-  cron: 0 8 * * 2
+- as: control-plane-249nodes
+  cron: 0 8 2 * *
```
Comment (afcollins) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:35]: Runs once a month on 2nd day. Correct?

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 10]
```diff
@@ -123,6 +123,42 @@ tests:
...
+    cluster_profile: aws-perfscale
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.xlarge
+      ENABLE_AUTOSCALING: "false"
+      MULTI_AZ: "true"
+      OPENSHIFT_VERSION: "4.14"
+      PROFILE_TYPE: reporting
+      READY_WAIT_TIMEOUT: 30m
+      REPLICAS: "249"
```
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:139]: Can we add PROFILE_TYPE: reporting

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml [hunk 11]
```diff
@@ -123,6 +123,42 @@ tests:
...
+    cluster_profile: aws-perfscale
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.xlarge
+      ENABLE_AUTOSCALING: "false"
+      MULTI_AZ: "true"
+      OPENSHIFT_VERSION: "4.14"
+      PROFILE_TYPE: reporting
+      READY_WAIT_TIMEOUT: 30m
+      REPLICAS: "120"
```
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.14-nightly-x86.yaml:157]: Can we add PROFILE_TYPE: reporting

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml [hunk 12]
```diff
@@ -118,6 +120,44 @@ tests:
...
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.xlarge
+      ENABLE_AUTOSCALING: "false"
+      MULTI_AZ: "true"
+      OPENSHIFT_VERSION: "4.15"
+      PODS_PER_NODE: "200"
+      PROFILE_TYPE: reporting
+      READY_WAIT_TIMEOUT: 30m
+      REPLICAS: "249"
```
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:137]: Can we add PROFILE_TYPE: reporting

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml [hunk 13]
```diff
@@ -118,6 +120,44 @@ tests:
...
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.xlarge
+      ENABLE_AUTOSCALING: "false"
+      MULTI_AZ: "true"
+      OPENSHIFT_VERSION: "4.15"
+      PODS_PER_NODE: "200"
+      PROFILE_TYPE: reporting
+      READY_WAIT_TIMEOUT: 30m
+      REPLICAS: "120"
```
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.15-nightly-x86.yaml:156]: Can we add PROFILE_TYPE: reporting

**Review by svetsa-rh — COMMENTED** [review]
Can we add PROFILE_TYPE: reporting for all these large scale tests, otherwise we will overload ES storage, mem and CPU.

**Review by afcollins — APPROVED** [review]
I looked through all the results.

The 4.14 cni 120 node error looks like what we were seeing with other clusters around that time so I consider it forgivable and not worth the cost for a re-run.

The other failure (4.15 cni 249 nodes) is also forgivable since the failing task (cerebus) failed due to timeout because kube-burner ran both metrics profiles. The PR ahs since changed reporting-only profile type. So again, the cost of a re-run is valuable since the kube-burner latencies are good.

/lgtm

**Review by afcollins — CHANGES_REQUESTED** [review]
I take that back.

Both 249node node-density jobs hit 2h task timeout.

```
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:169","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 2h0m0s timeout","severity":"error","time":"2024-01-25T23:31:22Z"}
```

[4.15](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/47827/rehearse-47827-periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.15-nightly-x86-control-plane-249nodes/1750684696016064512/build-log.txt)

[4.14](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/47827/rehearse-47827-periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-control-plane-249nodes/1750575169543344128/build-log.txt)

**Pull request closed without merging**